### DISCUSSION TITLE
Frequently asked questions about fossology Installation and working

### DISCUSSION THREAD

**Discussion by sritasngh** [fossology/fossology#1942] [Q&A]
This is Frequently asked questions section. If you don't get question which you are looking for then kindly post your question here as a new comment :)

**Comment by sritasngh** [comment]
`Question`: **How do I change my password?**

**Reply by sritasngh** [reply]
Go to menu ` Admin`, select sub menu `Users`, select sub menu item `Edit user account`.

**Comment by avinal** [comment]
`Question`: **My uploads are not visible by others, why?**

**Reply by sritasngh** [reply]
The uploads are visible per group, mostly depending on the visibility settings at upload. For each user also a per-user group is created (similar to unix / posix file systems with groups based access model). If you have selected your per-user group, where you are the only member, only you can see your uploads, clearing decisions, etc. In order to share your activities with others, please make sure that your main working group is selected. Or, at upload you can select another visibility level for this uploaded file.

**Comment by avinal** [comment]
`Question`: **I have installed from source, but cannot reach the FOSSology application under port 8081, what is wrong?**

**Reply by sritasngh** [reply]
At source install, the `Apache Web server` is installed on the host machine (the machine also FOSSology is installed on) and thus it needs to be called by the host, which is usually at `port 80`.

`Port 8081` is in use by docker and vagrant installations. In this installation FOSSology runs as guest in a virtualised environmet. For these environments, port 80 of the Apache Web server in the guest machine is mapped to port 8081 on the host, so it will not interfere with any deployed `http services` on port 80 on the host machine.

**Comment by avinal** [comment]
`Question`:**Sometimes the speed is very slow. It takes 10 minutes to generate the report?**

**Reply by sritasngh** [reply]
Because, `jobs` can be `scheduled` only at the number of available cores, subsequent jobs are in the queue. Normally, the generation of report generation will happen after some time.

**Comment by avinal** [comment]
`Question`:**Where can I find `users` table? How to drop the table?**

**Reply by sritasngh** [reply]
First Connect to PostgreSQL with 

``` bash 
psql -h localhost -d fossology -U fossy 
``` 

password: `fossy`.

To see the table use 
```bash
\dt
```
Now, to drop the users table use
```sql
DROP TABLE users CASCADE;
```

**Comment by avinal** [comment]
`Question`:**How can I test FOSSology without installing it every time after making changes in the source code?**

**Reply by sritasngh** [reply]
The FOSSology's design requires you to install every time you make changes. But Installation can be limited to a specific scope.
For example

- You have made a change in copyright agent
```bash
make build-copyright 
sudo make install-copyright
```

other steps can be skipped.

- You have made changes in UI to update the PHP code
```bash
sudo make install-ui empty-cache
```

- You have made changes in the DB
```bash 
# install everything first then run
sudo /usr/local/lib/fossology/fo-postinstall
```

**Comment by sritasngh** [comment]
`Question`: **How can I change the upload file size for fosslogy server?**

**Reply by sritasngh** [reply]
You can change it in your `php.ini`

> /etc/php/(php_version)/apache/php.ini

and change the values for 

> post_max_size and upload_max_filesize

After restarting the apache, it should accept new upload size limits.

**Reply by GMishx** [reply]
There is also a helper script: https://github.com/fossology/fossology/blob/master/install/scripts/php-conf-fix.sh

It changes the default upload limit to 700 mb, memory limit to 702 mb, increases the execution time to 5 minutes and changes the timezone to match machine's timezone.

**Reply by kartikkumar783** [reply]
Hi,Can I Increase the upload limit to more than 700MB ? If Yes , then how can I have installed fossology through Docker

**Reply by GMishx** [reply]
Yes, the limit can be increased. To do it, you can either edit the `php.ini` in Docker container with new values for `post_max_size` and `upload_max_filesize` and restart the container.

Or, the [install/scripts/php-conf-fix.sh](https://github.com/fossology/fossology/blob/master/install/scripts/php-conf-fix.sh) file can be edited with appropriate values for following parameters and a new image be built from it (note that `\1` is a regex place holder and not to be confused with `1702` numeric value).

https://github.com/fossology/fossology/blob/c1d7765e3af3327de71ee60991f1dc429c10c7c0/install/scripts/php-conf-fix.sh#L33-L37

**Reply by kartikkumar783** [reply]
@GMishx  How can I edit this file php-conf-fix.sh as i am using on docker

**Reply by kartikkumar783** [reply]
@GMishx Or How can I edit php.ini in docker container

**Reply by GMishx** [reply]
You can edit the `php-conf-fix.sh` in source and rebuild the Docker image. Then use it to create a new container......
You can also try [docker exec](https://docs.docker.com/engine/reference/commandline/exec/) to get inside the container, install vim/nano and edit the file.

**Reply by kartikkumar783** [reply]
@GMishx But I have pulled fossology through docker then how can I edit?

**Reply by GMishx** [reply]
Please have a look at above mentioned reference to docker-exec

**Comment by sritasngh** [comment]
`Question`: Can we customize the report generated by FOSSology?

**Reply by sritasngh** [reply]
Yes, FOSSology does not put any constraint on the generated documents.

**Reply by GMishx** [reply]
Also, most parts of the report can be configured by going to `Admin` then `Customize` and change the "Report Header Text" and other obligation fields.

Same can be done per upload by going to `Conf` section (link in the yellow bar while browsing an upload) and updating the `Unified Report Settings` tab.

**Comment by sritasngh** [comment]
`Question`: **How can I use CLI in FOSSology?**

**Reply by avinal** [reply]
There are many FOSSology CLI tools. You can upload, add jobs, check jobs, check licenses, and more right from the terminal. You can use the following small command snippet to list all the available CLI tools. Basically whatever starts with `fo_` or `foss` will most probably be a FOSSology CLI. 
```bash
ls /usr/local/bin | grep -E 'fo_|foss'
```

There is no existing documentation for all the tools, but you can use `--help` flag to list usage and available options. 
```bash
sudo fossupload_status --help
```
In general all the CLI tools needs `--username` (*default: fossy*) and `--password` (*default: fossy*) argument to be passed.
For example, to upload a file from version control, the following command can be used:
```bash
sudo cp2foss --username fossy --password fossy -G https://github.com/fossology/fossology.git
```

**Comment by sritasngh** [comment]
`Question`: I added a new `custom license` and gave it a `short name` that included `LicenseRef-` Now that I generated an `SPDX` report, I noticed that that `prefix` gets added automatically, so my license now has `LicenseRef-LicenseRef-` which is a bit ugly (even if most likely conformant with SPDX standard still). What does it mean?

**Reply by sritasngh** [reply]
FOSSology adds the prefix automatically if the license is not a `spdx` license.

**Comment by sritasngh** [comment]
`Question`: How can I change the `short name` of a license?

**Reply by sritasngh** [reply]
- You need to have `admin access` in order to change the short name of license already in use.
- The change of license short name from the `UI` is not allowed. 
- You need to connect to `DB` using 
```bash
psql -h localhost -p 5432 -Ufossy -dfossology 
UPDATE license_ref SET rf_shortname='correct-name' WHERE rf_shortname='LicenseRef-license';
```

- The default password for DB is `fossy`.

**Comment by sritasngh** [comment]
`Question`: Is there a way that I can upload a new license from `REST API` ?

**Reply by sritasngh** [reply]
You can `GET` the endpoint meaning you can search for licenses based on their `short name` which returns the `id`, `short name`, `full name` and `text`. Other operations are not yet supported.

**Comment by sritasngh** [comment]
`Question`:  What are the `OS` supported by FOSSology?

**Reply by sritasngh** [reply]
For now, FOSSology is not a `cross-platform` software system. It is best supported on `Ubuntu LTS` or `Debian GNU`.

**Reply by NicolasToussaint** [reply]
It can also be run in Docker containers.

**Reply by Aman-Codes** [reply]
Windows user's can also run Fossology software in [WSL](https://en.m.wikipedia.org/wiki/Windows_Subsystem_for_Linux)

**Comment by sritasngh** [comment]
`Question`: **How to delete a license added accidentally?**

**Reply by sritasngh** [reply]
You can't delete a license from `UI` but the license can be deactivated so that agents don't detect it. To do so follow:
- Go to `Admin` --> `License admin` --> `Select license`
- Find the license name and click on _edit_ to mark `active` as `NO`  

![Screenshot from 2021-05-13 23-00-23](https://user-images.githubusercontent.com/43489890/118163948-2ebd5a00-b440-11eb-9527-0ad74b58b561.png)

To delete it, you need to open the **fossology** database 
- Then find the license in `license_ref` and `license_candidate` tables. 
- You can search with column `rf_shortname` being the` shortname` or `rf_fullname`.
- Then delete the column by running SQL query

**Comment by QuattroN778** [comment]
`Question`:How do I disable an Agent?

**Comment by iamGohanHK** [comment]
Hi, I Installed fossology in my windows machine using docker WSL. 
docker pull fossology/fossology
docker run -p 8081:80 fossology/fossology


but localhos:8081/repo is not showing anything

**Comment by nishanthsankaran** [comment]
How do we go about creating a package name, version, and file URL, similar to the process of tagging a file for an open-source library?

**Comment by mridul45** [comment]
Hi I am unable to install fossology , everytime I try to build the project it says files  missing

**Reply by shaheemazmalmmd** [reply]
@mridul45 Can you add your log ??

**Comment by mridul45** [comment]
Kindly help

**Comment by mridul45** [comment]
HI @shaheemazmalmmd Here are the logs, I am trying to run the project since 2 days.

**Comment by mridul45** [comment]
![log1](https://github.com/fossology/fossology/assets/73380291/8d9925d2-4e2e-40f0-86b8-0aa411cc73e6)

**Comment by mridul45** [comment]
![log2](https://github.com/fossology/fossology/assets/73380291/50b1d4e2-050b-4d81-a33d-cea4fe6ada87)

**Comment by mridul45** [comment]
![log3](https://github.com/fossology/fossology/assets/73380291/395a97f3-2d61-4bb1-845e-5f0a94316ced)

**Comment by mridul45** [comment]
![log4](https://github.com/fossology/fossology/assets/73380291/7bbae326-9993-4052-a3c9-663eb7013796)

**Comment by mridul45** [comment]
![log5](https://github.com/fossology/fossology/assets/73380291/31e6735d-eecf-43da-ac6e-a2f840a49c72)

**Comment by mridul45** [comment]
![log6](https://github.com/fossology/fossology/assets/73380291/a54a8e90-989b-464d-8f8e-f564a6b17694)
